### PR TITLE
Session ID Change

### PR DESCRIPTION
**Pull request by lgenet**
**Description:**
Changed out the session cookie ID for sessions (confusing name)  this changes the session_id property of the cookies so that they are cms_tn_session_id.  This is because other sites were overriding our session id and causing csrf errors.

### CODE REVIEW

File: include/session/session.js [hunk 1]
```diff
@@ -296,16 +296,14 @@ module.exports = function SessionModule(pb) {
...
-                if(key.trim() == 'session_id'){
-                    sessionId = request.headers[SessionHandler.COOKIE_HEADER][key];
-                    break;
-                }
+        let sessionId = null;
+
+        let cookieHeader = request.headers[SessionHandler.COOKIE_HEADER];
+        if (cookieHeader) {
+            let key = Object.keys(cookieHeader).find(x => x.trim() === SessionHandler.COOKIE_NAME) ||
+                Object.keys(cookieHeader).find(x => x.trim() === 'session_id');
```
Comment (seanpaw) [inline include/session/session.js]: Is there a reason to iterate 2x here?

File: test/include/session/session_spec.js [hunk 2]
```diff
@@ -0,0 +1,160 @@
+const {expect, sinon} = require('../../helpers/spec_helper');
```
Comment (seanpaw) [inline test/include/session/session_spec.js:1]: testing++

File: include/session/session.js (hunk 1 above)
Comment (lgenet) [reply include/session/session.js]: yes!  While we are in the process of migration we must first check to see if they have the new cms_tn session id.  If not, we want to try and fall back to the old session_id.  We can remove this code after 24 hours when all sessions have expired.

File: test/include/session/session_spec.js (hunk 2 above)
Comment (lgenet) [reply test/include/session/session_spec.js:1]: I wish I could have done more, but I went beyond what I changed so that is an increase :)

Comment (lgenet) [reply test/include/session/session_spec.js:1]: also moving them out of the world of should and into the land of expect, sinon, mocha, chia, and such.  Not quite to jest but hey

**Pull request merged**